model/TestCenterService.php

```diff
@@ -201,4 +204,58 @@ public function getTestCenter($testCenterUri)
     {
         return new core_kernel_classes_Resource($testCenterUri);
     }
+
+    /**
+     * @param core_kernel_classes_Resource $testCenter
+     * @param User $user
+     * @param core_kernel_classes_Resource $role
+     * @return bool
+     * @throws TestCenterException
+     */
+    public function assignUser(\core_kernel_classes_Resource $testCenter, User $user, \core_kernel_classes_Resource $role)
+    {
+        $userResource = $this->getResource($user->getIdentifier());
+        $userRoles = $userResource->getPropertyValues($this->getProperty(UserRdf::PROPERTY_ROLES));
+
+        if (!in_array($role->getUri(), $userRoles)) {
+            throw new TestCenterException(__('User with given role cannot be assigned to the test center.'));
+        }
+
+        switch ($role->getUri()) {
+            case self::ROLE_TESTCENTER_ADMINISTRATOR:
+                $assignProperty = $this->getProperty(ProctorManagementService::PROPERTY_ADMINISTRATOR_URI);
+                break;
+            case ProctorService::ROLE_PROCTOR:
+                $assignProperty = $this->getProperty(ProctorManagementService::PROPERTY_ASSIGNED_PROCTOR_URI);
+                break;
+            default:
+                throw new TestCenterException(__('Role is not allowed.'));
+        }
+        return $userResource->editPropertyValues($assignProperty, $testCenter);
+    }
+
+    /**
+     * @param core_kernel_classes_Resource $testCenter
+     * @param User $user
+     * @param core_kernel_classes_Resource $role
+     * @return bool
+     * @throws TestCenterException
+     */
+    public function unassignUser(\core_kernel_classes_Resource $testCenter, User $user, \core_kernel_classes_Resource $role)
+    {
+        $userResource = $this->getResource($user->getIdentifier());
```

Comment:
I would move this line below the switch case to avoid hitting the database in case of exception. What do you think?

Reply:
`getResource` does not make queries to DB